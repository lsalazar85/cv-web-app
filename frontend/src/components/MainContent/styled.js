import styled from 'styled-components';
import { fadeIn } from '../../styles/animation';

const MainContentContainer = styled.main`
    display: flex;
    flex: 2;
    height: 100%;
    overflow: auto;
    scrollbar-width: thin;    
    scrollbar-color: rgba(136, 136, 136, 0.15) transparent;
    ${fadeIn({time: '400ms'})};

    &::-webkit-scrollbar {
        width:  5px;
    }

    &::-webkit-scrollbar-thumb {
        background: rgba(136, 136, 136, 0.15);
    }

    &::-webkit-scrollbar-track {
        background: transparent;
    }

    @media (min-width: 320px){
        background: #1c1c1c;
        padding: 15px;
    }

    @media (min-width: 992px){
        background: transparent;
        padding: 32px;
        max-width: 624px;
    }
`

export { MainContentContainer }