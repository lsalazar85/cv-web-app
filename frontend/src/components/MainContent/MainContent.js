import React from 'react'
import { Switch, Route } from 'react-router-dom'

import AboutMe from '../../pages/AboutMe'
import Contact from '../../pages/Contact'
import Education from '../../pages/Education'
import WorkExperience from '../../pages/WorkExperience'

import { MainContentContainer } from './styled'

const MainContent = () => (
  <Switch>
    <MainContentContainer>
      <Route exact path="/" render={() => <AboutMe />} />
      <Route path="/contact" render={() => <Contact />} />
      <Route path="/work-experience" render={() => <WorkExperience />} />
      <Route path="/education" render={() => <Education />} />
    </MainContentContainer>
  </Switch>
)

export default MainContent
