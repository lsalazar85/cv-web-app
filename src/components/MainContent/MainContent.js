import React from 'react';
import { Switch, Route } from 'react-router-dom';

import AboutMe from '../AboutMe';
import InfoPersonal from '../InfoPersonal';
import Education from '../Education';
import WorkExperience from '../WorkExperience';

import { MainContentContainer } from './styled';

const MainContent = () => (
    <Switch>
        <MainContentContainer>
            <Route exact path="/" render={() => <AboutMe />}/>
            <Route path="/personal-info" render={() => <InfoPersonal />}/>
            <Route path="/work-experience" render={() => <WorkExperience />}/>
            <Route path="/education" render={() => <Education />}/>                                   
        </MainContentContainer>
    </Switch>  
)

export default MainContent