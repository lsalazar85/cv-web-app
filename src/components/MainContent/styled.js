import styled from 'styled-components';

const MainContentContainer = styled.div`
    display: flex;
    flex-direction: column;
    flex-shrink: 0;
    flex-grow: 1;
`

export { MainContentContainer }