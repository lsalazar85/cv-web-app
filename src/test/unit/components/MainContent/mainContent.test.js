import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import MainContent from '../../../../components/MainContent';

describe('MainContent', () => {
    afterEach(cleanup);

    test('should render correctly', () => {
        const { asFragment } = render(
          <Router>
            <MainContent />
          </Router>  
        );
        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    });

})