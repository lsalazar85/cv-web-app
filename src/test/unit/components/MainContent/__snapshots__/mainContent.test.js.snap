// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent check render & navigating to contact 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kVdUYB"
  >
    <div>
      Contact
    </div>
  </div>
</DocumentFragment>
`;

exports[`MainContent check render & navigating to education 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kVdUYB"
  >
    <div>
      Education
    </div>
  </div>
</DocumentFragment>
`;

exports[`MainContent check render & navigating to home 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kVdUYB"
  >
    <div>
      AboutMe
    </div>
  </div>
</DocumentFragment>
`;

exports[`MainContent check render & navigating to work experience 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kVdUYB"
  >
    <div />
  </div>
</DocumentFragment>
`;
